src/notebookTemplates/listNotebookTemplates.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { useEffect, useState } from 'react';
+import { useTable, useGlobalFilter, usePagination } from 'react-table';
+import { LabIcon } from '@jupyterlab/ui-components';
+import filterIcon from '../../style/icons/filter_icon.svg';
+import { ClipLoader } from 'react-spinners';
+import GlobalFilter from '../utils/globalFilter';
+import TableData from '../utils/tableData';
+import { PaginationView } from '../utils/paginationView';
+import { ICellProps, loggedFetch } from '../utils/utils';
+import * as path from 'path';
+import { NOTEBOOK_TEMPLATES_LIST_URL } from '../utils/const';
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { IDefaultFileBrowser } from '@jupyterlab/filebrowser';
+import { IThemeManager } from '@jupyterlab/apputils';
+
+function ListNotebookTemplates({
+  app,
+  themeManager,
+  defaultFileBrowser
+}: {
+  app: JupyterFrontEnd;
+  themeManager: IThemeManager;
+  defaultFileBrowser: IDefaultFileBrowser;
+}) {
+  const iconFilter = new LabIcon({
+    name: 'launcher:filter-icon',
+    svgstr: filterIcon
+  });
+
+  const [templateList, setTemplateList] = useState<any[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const data = templateList;
+  const columns = React.useMemo(
+    () => [
+      {
+        Header: 'Category',
+        accessor: 'category'
+      },
+      {
+        Header: 'Name',
+        accessor: 'name'
+      },
+      {
+        Header: 'Description',
+        accessor: 'description'
+      },
+      {
+        Header: '',
+        accessor: 'actions'
+      }
+    ],
+    []
+  );
+  const downloadNotebook = async (
+    notebookContent: any,
+    notebookUrl: string
+  ) => {
+    const contentsManager = app.serviceManager.contents;
+
+    // Define the path to the 'notebookTemplateDownload' folder within the local application directory
+    const notebookTemplateDownloadFolderPath = defaultFileBrowser.model.path;
+
+    const urlParts = notebookUrl.split('/');
+    const filePath = `${notebookTemplateDownloadFolderPath}${path.sep}${
+      urlParts[urlParts.length - 1]
+    }`;
+
+    // Save the file to the workspace
+    await contentsManager.save(filePath, {
+      type: 'file',
+      format: 'text',
+      content: JSON.stringify(notebookContent)
+    });
+
+    // Refresh the file fileBrowser to reflect the new file
+    app.shell.currentWidget?.update();
+
+    app.commands.execute('docmanager:open', {
+      path: filePath
+    });
+  };
+
+  const handleClick = async (template: any) => {
+    const notebookUrl = template.url;
+
+    loggedFetch(notebookUrl, {
```

Comment:
Move all the API calls to service files.

Reply:
Moved the calls to service file